### PR TITLE
TST: adding new test for groupby cumsum with named aggregate

### LINKED ISSUE #41720: BUG: not possible to 'cumsum' Timedelta with named aggregation
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of pandas.

- [ ] (optional) I have confirmed this bug exists on the master branch of pandas.

---

#### Code Sample, a copy-pastable example

```python
import pandas as pd
import numpy as np
import random

# Dataset
ts = pd.DatetimeIndex([pd.Timestamp('2021/01/01 00:30'),
                       pd.Timestamp('2021/01/01 00:45'),
                       pd.Timestamp('2021/01/01 02:00'),
                       pd.Timestamp('2021/01/01 03:50'),
                       pd.Timestamp('2021/01/01 05:00')])
length = len(ts)
random.seed(1)
value = random.sample(range(1, length+1), length)
df = pd.DataFrame({'value': value, 'ts': ts})
df['td'] = df['ts'] - df['ts'].shift(1, fill_value=ts[0]-pd.Timedelta('1h'))
df['amount'] = df['value']*10
df.loc[:2,'grps'] = 'a'
df.loc[2:,'grps'] = 'b'
```
```python
In [16]: df
Out[16]: 
   value                  ts              td  amount grps
0      2 2021-01-01 00:30:00 0 days 01:00:00      20    a
1      1 2021-01-01 00:45:00 0 days 00:15:00      10    a
2      5 2021-01-01 02:00:00 0 days 01:15:00      50    b
3      4 2021-01-01 03:50:00 0 days 01:50:00      40    b
4      3 2021-01-01 05:00:00 0 days 01:10:00      30    b
```
```python
# cumsum and Timedelta work when used through pd.Series
df['td'].cumsum()

# cumsum and Timedelta do not work when used in groupby with named aggregation
agg_rules = {'td':('td', 'cumsum')}
res = df.groupby('grps').agg(**agg_rules)
```
Error message that is returned
```python
res = df.groupby('grps').agg(**agg_rules)
Traceback (most recent call last):

  File "<ipython-input-14-c0f4606d675a>", line 2, in <module>
    res = df.groupby('grps').agg(**agg_rules)

  File "/home/pierre/anaconda3/lib/python3.8/site-packages/pandas/core/groupby/generic.py", line 945, in aggregate
    result, how = aggregate(self, func, *args, **kwargs)

  File "/home/pierre/anaconda3/lib/python3.8/site-packages/pandas/core/aggregation.py", line 582, in aggregate
    return agg_dict_like(obj, arg, _axis), True

  File "/home/pierre/anaconda3/lib/python3.8/site-packages/pandas/core/aggregation.py", line 768, in agg_dict_like
    results = {key: obj._gotitem(key, ndim=1).agg(how) for key, how in arg.items()}

  File "/home/pierre/anaconda3/lib/python3.8/site-packages/pandas/core/aggregation.py", line 768, in <dictcomp>
    results = {key: obj._gotitem(key, ndim=1).agg(how) for key, how in arg.items()}

  File "/home/pierre/anaconda3/lib/python3.8/site-packages/pandas/core/groupby/generic.py", line 247, in aggregate
    ret = self._aggregate_multiple_funcs(func)

  File "/home/pierre/anaconda3/lib/python3.8/site-packages/pandas/core/groupby/generic.py", line 315, in _aggregate_multiple_funcs
    results[base.OutputKey(label=name, position=idx)] = obj.aggregate(func)

  File "/home/pierre/anaconda3/lib/python3.8/site-packages/pandas/core/groupby/generic.py", line 241, in aggregate
    return getattr(self, func)(*args, **kwargs)

  File "/home/pierre/anaconda3/lib/python3.8/site-packages/pandas/core/groupby/groupby.py", line 2497, in cumsum
    return self._cython_transform("cumsum", **kwargs)

  File "/home/pierre/anaconda3/lib/python3.8/site-packages/pandas/core/groupby/groupby.py", line 996, in _cython_transform
    raise DataError("No numeric types to aggregate")

DataError: No numeric types to aggregate
```

#### Problem description

Cumsum works with Timedelta in some workflows, but for some other workflows (groupby, named aggregation) it does not.

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : 2cb96529396d93b46abab7bbc73a208e708c642e
python           : 3.8.5.final.0
python-bits      : 64
OS               : Linux
OS-release       : 5.8.0-53-generic
Version          : #60~20.04.1-Ubuntu SMP Thu May 6 09:52:46 UTC 2021
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : fr_FR.UTF-8
LOCALE           : fr_FR.UTF-8

pandas           : 1.2.4
numpy            : 1.20.1
pytz             : 2021.1
dateutil         : 2.8.1
pip              : 21.1.1
setuptools       : 52.0.0.post20210125
Cython           : 0.29.23
pytest           : 6.2.3
hypothesis       : None
sphinx           : 4.0.1
blosc            : None
feather          : None
xlsxwriter       : 1.3.8
lxml.etree       : 4.6.3
html5lib         : 1.1
pymysql          : None
psycopg2         : None
jinja2           : 3.0.0
IPython          : 7.22.0
pandas_datareader: None
bs4              : 4.9.3
bottleneck       : 1.3.2
fsspec           : 0.9.0
fastparquet      : 0.6.3
gcsfs            : None
matplotlib       : 3.3.4
numexpr          : 2.7.3
odfpy            : None
openpyxl         : 3.0.7
pandas_gbq       : None
pyarrow          : 2.0.0
pyxlsb           : None
s3fs             : None
scipy            : 1.6.2
sqlalchemy       : 1.4.15
tables           : 3.6.1
tabulate         : None
xarray           : None
xlrd             : 2.0.1
xlwt             : 1.3.0
numba            : 0.51.2

</details>

### PR DESCRIPTION
**Pull request by seanjedi**
- [ ] closes #41720 
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.

### CODE REVIEW

File: pandas/tests/extension/base/groupby.py [hunk 1]
```diff
@@ -56,6 +56,39 @@ def test_groupby_agg_extension(self, data_for_grouping):
...
+                pd.Timestamp("2021/01/01 02:00"),
+            ]
+        )
+        df = pd.DataFrame({"value": [1, 2, 3], "ts": ts})
+        df["td"] = df["ts"] - df["ts"].shift(1, fill_value=ts[0] - pd.Timedelta("1h"))
+        df["amount"] = df["value"] * 10
+        df.loc[:2, "grps"] = "a"
+        df.loc[2:, "grps"] = "b"
+        agg_rules = {"td": ("td", "cumsum")}
+        result = df.groupby("grps").agg(**agg_rules)
```
Comment (rhshadrach) [inline pandas/tests/extension/base/groupby.py:89]: can you combine these two as `.agg(td=("td", "cumsum"))`

File: pandas/tests/extension/base/groupby.py [hunk 2]
```diff
@@ -56,6 +56,39 @@ def test_groupby_agg_extension(self, data_for_grouping):
...
+                        2: pd.Timedelta("0 days 01:15:00"),
+                    }
+                }
+            )
+        ],
+    )
+    def test_groupby_agg_extension_timedelta_cumsum_with_named_aggregation(
+        self, expected
+    ):
+        ts = pd.DatetimeIndex(
```
Comment (rhshadrach) [inline pandas/tests/extension/base/groupby.py:76]: Add the GH issue number as a comment on the first line of the test; `# GH#41720` here.

**Review by rhshadrach — CHANGES_REQUESTED** [review]
Looking good! Some requests below. In general, you don't need to use the exact code that's in the issue; it's best to clean up and simplify the example in the issue when possible.

File: pandas/tests/extension/base/groupby.py (hunk 1 above)
Comment (rhshadrach) [reply pandas/tests/extension/base/groupby.py:89]: Also, this is a bit of a nit pick, but it is best to separate out the call to `groupby` from the call to `agg`. These makes setting breakpoints easier for when developers have to debug later on.

```
gb = df.groupby("grps")
result = gb.agg(td=("td", "cumsum"))
```

File: pandas/tests/extension/base/groupby.py [hunk 3]
```diff
@@ -56,6 +56,39 @@ def test_groupby_agg_extension(self, data_for_grouping):
         result = df.groupby("A").first()
         self.assert_frame_equal(result, expected)
 
+    @pytest.mark.parametrize(
+        "expected",
```
[the remark this answers is not in the document]
Comment (seanjedi) [reply pandas/tests/extension/base/groupby.py]: Sure, This has been updated in: d12629013a9e68981b7715bd0f53d8851f8bdb0f

File: pandas/tests/extension/base/groupby.py [hunk 4]
```diff
@@ -56,6 +56,39 @@ def test_groupby_agg_extension(self, data_for_grouping):
...
+                pd.Timestamp("2021/01/01 00:30"),
+                pd.Timestamp("2021/01/01 00:45"),
+                pd.Timestamp("2021/01/01 02:00"),
+            ]
+        )
+        df = pd.DataFrame({"value": [1, 2, 3], "ts": ts})
+        df["td"] = df["ts"] - df["ts"].shift(1, fill_value=ts[0] - pd.Timedelta("1h"))
+        df["amount"] = df["value"] * 10
+        df.loc[:2, "grps"] = "a"
+        df.loc[2:, "grps"] = "b"
```
[the remark this answers is not in the document]
Comment (seanjedi) [reply pandas/tests/extension/base/groupby.py]: It seems to be possible, I believe this should work: d12629013a9e68981b7715bd0f53d8851f8bdb0f

File: pandas/tests/extension/base/groupby.py (hunk 1 above)
Comment (seanjedi) [reply pandas/tests/extension/base/groupby.py]: Sure, this seems reasonable to me! 
I have updated the code here: d12629013a9e68981b7715bd0f53d8851f8bdb0f

File: pandas/tests/extension/base/groupby.py (hunk 2 above)
Comment (seanjedi) [reply pandas/tests/extension/base/groupby.py]: Sure, I have added that here: d12629013a9e68981b7715bd0f53d8851f8bdb0f

File: pandas/tests/extension/base/groupby.py [hunk 5]
```diff
@@ -56,6 +56,40 @@ def test_groupby_agg_extension(self, data_for_grouping):
...
+            {
+                "value": pd.Series(["1", "2", "3"], dtype="int64"),
+                "ts": pd.Series(
+                    [
+                        "2021-01-01 00:30:00",
+                        "2021-01-01 00:45:00",
+                        "2021-01-01 02:00:00",
+                    ],
+                    dtype="datetime64[ns]",
+                ),
```
Comment (rhshadrach) [inline pandas/tests/extension/base/groupby.py:80]: Looks like value, ts, and amount (below) can be removed?

File: pandas/tests/extension/base/groupby.py [hunk 6]
```diff
@@ -56,6 +56,40 @@ def test_groupby_agg_extension(self, data_for_grouping):
...
+                "td": pd.Series(
+                    ["0 days 01:00:00", "0 days 00:15:00", "0 days 01:15:00"],
+                    dtype="timedelta64[ns]",
+                ),
+                "amount": pd.Series(["10", "20", "30"], dtype="int64"),
+                "grps": pd.Series(["a", "a", "b"], dtype="object"),
```
Comment (rhshadrach) [inline pandas/tests/extension/base/groupby.py:86]: Can just specify as a list here: `"grps": ["a", "a", "b"],`

**Review by rhshadrach — CHANGES_REQUESTED** [review]

File: pandas/tests/extension/base/groupby.py (hunk 5 above)
Comment (seanjedi) [reply pandas/tests/extension/base/groupby.py]: Hmm, I believe you are right, I have simplified that here: 6d8ae022355020b73c98e22c216381b81569f0b6

File: pandas/tests/extension/base/groupby.py (hunk 6 above)
Comment (seanjedi) [reply pandas/tests/extension/base/groupby.py]: Ohh alright, I wasn't entirely sure, but checking it, it seems to work, so I updated it here: 6d8ae022355020b73c98e22c216381b81569f0b6

**Review by rhshadrach — APPROVED** [review]
lgtm

**Comment by rhshadrach** [comment]
Thanks @seanjedi!

**Pull request merged**